docs/source/blueprints/BP005.rst

```diff
@@ -0,0 +1,275 @@
+central repository.
+
+However, if the utility intends to ease the interaction with an esoteric
+feature, system or device, this may not be a good place for it.
+
+As a general rule, if a system level feature or interface is generically
+available to a multitude of users, such as the partition example above, it is a
+good candidate for this repository. Common sense is much appreciated here.
+
+Besides that, the proposed utility must have at least one MAINTAINER that will
+be responsible for keeping track of Pull Requests, tests and quality
+requirements as described below:
+
+Quality requirements
+--------------------
+
+It’s expected that both new modules and changes to existing modules will need
+to observe a full set of quality requirements and general standards that will
+either be described (documentation) and enforced as automated checks as much as
+possible.
+
+Among the topics related to the quality requirements and standards, the
+contribution documentation and automated checks will cover, at least:
+
+* Logical, stylistic, analytical and formatter checks
+
+  * Docstring checks
+  * Pylint
+  * PEP-8, PEP-257, etc…
+
+* Unit and Functional Tests Coverage
+
+  Due to the current code quality and given this is a new repository we would
+  like to reach at least 90% of coverage. This way, users writing tests relying
+  on these libraries will have a guarantee of a good level of correctness and
+  quality, redirecting the debugging and troubleshooting efforts away from the
+  well tested libraries.
+
+The new standalone repository will have predefined pipelines to run those
+checks, so new libraries will be automatically enforced to newly introduced
+libraries.
+
+Metadata description
+--------------------
+
+Every module will need to provide a description including:
+
+* Utility description
+* The utility category (ie: network, cpu, security, etc..)
+* Maintainers information
+  * Main contact (We need a contact point, a single maintainer or a group)
+  * Maintainer Group members (this will be used to block merges)
+* Supported Platforms
+* Supported execution method (Local vs Remote)
+
+For instance, the textual category and description will allow users to search
+for modules based on their interest.  Also, our pipelines will parse that data
+field containing the list of supported platforms, and will trigger tests on
+those platforms. Users will then be able to see the results of those tests on
+each of the platforms defined, giving them quality assurance and coverage
+there.
+
+The exact format of the metadata file is still to be defined, but a possibility
+is to define a JSON schema that will be provided on this new repository.
+
+Maintainability
+===============
+
+As described above, each library must follow the new standard and keep that
+same level for future versions. Because of that, new PRs must be approved by at
+least one MAINTAINER and pass the basic checks pipelines that will be provided
```

Comment:
@beraldoleal @clebergnu module wise MAINTAINER/s is really a good thought.
want to hear more on roles and authorities of MAINTAINER/s here. 
is this the same what we have currently or added with any axtra responsibilities.

Reply:
Indeed, this is where Avocado VT has tried to be for years now but never went formally. At the moment lots of different areas of code there have 1 or less maintainers that everybody pings when respective changes are introduced by a PR and we all do this implicitly just from memory. There is a MAINTAINERS file there but it points to general maintainers without any finer resolution on whom to ping about a particular change. In this regard I am happy to see this repo and blueprint trying to amend what has been a long-standing problem there and one that could often freeze a lot of PR-s indefinitely and discourage contribution of any kind.

In summary: I really like this idea and it scales a lot better for larger teams, of course assuming that there is sufficiently strong automation to iron out most of the difference in style and approach among contributors and maintainers.

Reply:
>  is this the same what we have currently or added with any axtra responsibilities.

Probably it will have more or less what we have today, but it will be formal and well defined. Maintainers would be responsible not only of its plugin but also make sure it is following the general guide lines to make sure its plugin will not divert much from the others. Maintainers, as described here in this blueprint, will care mainly about maintainability (including quality), releases and discussions involving this new repository.

> In summary: I really like this idea and it scales a lot better for larger teams, of course assuming that there is sufficiently strong automation to iron out most of the difference in style and approach among contributors and maintainers.

Good! :) Since those plugins will be part of the same repository, they will share a common contributor guide, automation tools and pipelines. So in terms of logical, stylistic, analytical and formatter checks they could share the same rules. For sure, those rules must be agreed by us when drafting the contributor guide there.

Reply:
@clebergnu I'll be glad, thank you!